pkg/scheduler/framework/plugins/podtopologyspread/filtering_test.go

```diff
@@ -1450,12 +1534,66 @@ func TestSingleConstraint(t *testing.T) {
 				"node-y": framework.Unschedulable,
 			},
 		},
+		{
+			name: "pods spread across nodes as 2/2, maxSkew is 2, and the number of domains < minDomains, then no node fits",
+			pod: st.MakePod().Name("p").Label("foo", "").SpreadConstraint(
+				2,
+				"node",
+				v1.DoNotSchedule,
+				st.MakeLabelSelector().Exists("foo").Obj(),
+				pointer.Int32(3), // larger than the number of domains(2)
+			).Obj(),
+			nodes: []*v1.Node{
+				st.MakeNode().Name("node-a").Label("node", "node-a").Obj(),
+				st.MakeNode().Name("node-b").Label("node", "node-b").Obj(),
+			},
+			existingPods: []*v1.Pod{
+				st.MakePod().Name("p-a1").Node("node-a").Label("foo", "").Obj(),
+				st.MakePod().Name("p-a2").Node("node-a").Label("foo", "").Obj(),
+				st.MakePod().Name("p-b1").Node("node-b").Label("foo", "").Obj(),
+				st.MakePod().Name("p-b2").Node("node-b").Label("foo", "").Obj(),
```

Comment:
You can just have one pod on node-b, so in the expected result, node-a is Unschedulable, node-b is Success.